packages/esm-patient-search-app/src/patient-search-page/patient-search-lg.component.tsx

```diff
@@ -0,0 +1,151 @@
+import PatientBanner, { PatientBannerSkeleton } from './patient-banner/banner/patient-banner.component';
+import Pagination from '../ui-components/pagination/pagination.component';
+import { usePatientSearchPaginated } from '../patient-search.resource';
+import { interpolateString, navigate, useConfig } from '@openmrs/esm-framework';
+
+interface PatientSearchComponentProps {
+  query: string;
+  inTabletOrOverlay?: boolean;
+  stickyPagination?: boolean;
+  onPatientSelect?: (patientUuid: string) => void;
+}
+
+const PatientSearchComponent: React.FC<PatientSearchComponentProps> = ({
+  query,
+  stickyPagination,
+  onPatientSelect,
+  inTabletOrOverlay,
+}) => {
+  const { t } = useTranslation();
+  const config = useConfig();
+  const [currentPage, setPage] = useState(1);
+  const resultsToShow = inTabletOrOverlay ? 15 : 5;
+  const {
+    isLoading,
+    data: searchResults,
+    fetchError,
+    hasMore,
+    totalResults,
+  } = usePatientSearchPaginated(query, !!query, resultsToShow, currentPage);
+
+  const totalPages = useMemo(
+    () => (totalResults ? Math.ceil(totalResults / resultsToShow) : 0),
+    [totalResults, resultsToShow, Math.ceil],
+  );
+
+  const handlePatientSelection = useCallback(
+    (evt, patientUuid: string) => {
+      evt.preventDefault();
+      if (onPatientSelect) {
+        onPatientSelect(patientUuid);
+      } else {
+        navigate({
+          to: interpolateString(config.search.patientResultUrl, {
+            patientUuid: patientUuid,
+          }),
+        });
+      }
+    },
+    [navigate, interpolateString, config, onPatientSelect],
+  );
+
+  if (!query) {
+    return <></>;
+  }
+
+  if (isLoading) {
+    return (
+      <div className={styles.searchResultsContainer}>
+        <div className={`${styles.searchResults} ${stickyPagination && styles.broadBottomMargin}`}>
+          <div>
+            <h2
+              className={`${styles.resultsHeader} ${styles.paddedResultsHeader} ${
+                inTabletOrOverlay && styles.leftPaddedResultHeader
+              }`}>
+              <span className={styles.productiveHeading02}>
+                {totalResults
+                  ? `${totalResults} ${t('seachResultsSmall', 'search results')}`
+                  : t('searchingText', 'Searching...')}
+              </span>
+            </h2>
+            <div className={styles.results}>
+              <PatientBannerSkeleton />
+              <PatientBannerSkeleton />
+              <PatientBannerSkeleton />
+              <PatientBannerSkeleton />
+              <PatientBannerSkeleton />
+            </div>
+          </div>
+          {totalResults !== 1 && (
+            <div className={`${styles.pagination} ${stickyPagination && styles.stickyPagination}`}>
+              <Pagination setPage={setPage} currentPage={currentPage} hasMore={hasMore} totalPages={totalPages} />
+            </div>
+          )}
+        </div>
+      </div>
+    );
+  }
+
+  return !fetchError ? (
```

Comment:
These nested ternaries are a little hard to read. Can you reformat this to be more declarative? E.g.
```
if (fetchError) { return ...} 
if (!fetchError && isEmpty(searchResults)) { return ...} 
```